qiskit/qasm2/parse.py

```diff
@@ -318,6 +321,9 @@ def _define(self):
                 raise ValueError(f"received invalid bytecode to build gate: {op}")
         self._definition = qc
 
+    def __array__(self, dtype=None):
```

Comment:
This will need a copy kwarg for numpy 2.0 support right?

Reply:
Yeah, I wrote this PR before the Numpy 2.0 one

Reply:
Should be done in 88ff0d6a90.